src/Native/System.Net.Security.Native/pal_gssapi.cpp

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "pal_gssapi.h"
+#include <assert.h>
+#include <string.h>
+
+static void NetSecurity_HandleError(uint32_t majorStatus, gss_buffer_t *outBufferHandle, uint32_t *outBufferLength)
+{
+    if (GSS_ERROR(majorStatus) || *outBufferHandle == nullptr || (*outBufferHandle)->length > UnsignedCast(std::numeric_limits<int>::max()))
+    {
+        uint32_t relBufferStatus;
+        gss_release_buffer(&relBufferStatus, *outBufferHandle);
+        *outBufferHandle = nullptr;
+        *outBufferLength = 0;
+    }
+    else
+    {
+        *outBufferLength = static_cast<uint32_t>((*outBufferHandle)->length);
```

Comment:
> Why not make the parameter int64_t then?
Actually this parameter corresponds to the length of a managed array. So at some point, we need to do the conversion. Do you prefer that the casting happens at the managed layer?